tests/cpp/common/test_json.cc

```diff
@@ -545,7 +545,34 @@ TEST(Json, RoundTrip) {
     }
 
     auto t = i;
-    i+= static_cast<uint32_t>(dist(&rng));
+    i += static_cast<uint32_t>(dist(&rng));
+    if (i < t) {
+      break;
+    }
+  }
+}
+
+TEST(Json, DISABLED_RoundTripExhaustive) {
+  uint32_t i = 0;
+  SimpleLCG rng;
+  SimpleRealUniformDistribution<float> dist(1.0f, 4096.0f);
+
+  while (i <= std::numeric_limits<uint32_t>::max()) {
+    float f;
+    std::memcpy(&f, &i, sizeof(f));
+
+    Json jf { f };
+    std::string str;
+    Json::Dump(jf, &str);
+    auto loaded = Json::Load({str.c_str(), str.size()});
+    if (XGBOOST_EXPECT(std::isnan(f), false)) {
+      ASSERT_TRUE(std::isnan(get<Number const>(loaded)));
+    } else {
+      ASSERT_EQ(get<Number const>(loaded), f);
+    }
+
+    auto t = i;
+    ++i;
     if (i < t) {
       break;
     }
```

Comment:
No longer needed.  You can't overflow the integer when adding 1.

Reply:
Actually we still need it, since we run the loop until `i` becomes equal to `std::numeric_limits<uint32_t>::max()`. Then running `++i` would set `i` to zero.

Reply:
Sure, you can change `<=` to `<`.  Dropping a branch.